sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_client_async.py

```diff
@@ -208,7 +209,29 @@ async def _do_retryable_operation_async(self, operation, *args, **kwargs):
                         break
                     await asyncio.sleep(self._retry_policy.get_backoff_time(retry_settings, exc))
                     if exc.condition == ErrorCondition.LinkDetachForced:
+                        total_link_credit = self._link.total_link_credit
                         await self._close_link_async()  # if link level error, close and open a new link
+                        if isinstance(self._link, SenderLink):
+                            self._link = self._session.create_sender_link(
+                                target_address=self.target,
+                                link_credit=self._link_credit,
+                                send_settle_mode=self._send_settle_mode,
+                                rcv_settle_mode=self._receive_settle_mode,
+                                max_message_size=self._max_message_size,
+                                properties=self._link_properties,
```

Comment:
do we need to pop the self._link off of the input_handles[] in session.py, or does it reuse the same input handle indices when reattaching?

Reply:
if we want to reattach the same link we dont want to pop the input handles I don't think